src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,101 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private int size;
+    private int threshold = (int) (INITIAL_CAPACITY * LOAD_FACTOR);
+    private Node<K, V>[] table = (Node<K, V>[]) new Node[INITIAL_CAPACITY];
+
+    private static class Node<K, V> {
+        private final int hash;
+        private final K key;
+        private V value;
+        private Node<K, V> next;
+
+        Node(int hash, K key, V value, Node<K, V> next) {
+            this.hash = hash;
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= threshold) {
+            resize();
+        }
+        putVal(key, value, table);
     }
 
     @Override
     public V getValue(K key) {
+        Node<K, V> currentNode = table[hash(key) % table.length];
+        while (currentNode != null) {
+            if (Objects.equals(currentNode.key, key)) {
+                return currentNode.value;
+            }
+            currentNode = currentNode.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int hash(Object key) {
+        return (key == null) ? 0 : (31 * 17 + Math.abs(key.hashCode()));
+    }
+
+    private void putVal(K key, V value, Node<K, V>[] tab) {
```

Comment:
Why you do not use table?
```suggestion
    private void putVal(K key, V value) {
```